sdk/core/src/main/java/com/klaviyo/coresdk/model/UserInfo.kt

```diff
@@ -0,0 +1,99 @@
+package com.klaviyo.coresdk.model
+
+import com.klaviyo.coresdk.Registry
+import com.klaviyo.coresdk.model.ProfileKey.ANONYMOUS
+import com.klaviyo.coresdk.model.ProfileKey.EMAIL
+import com.klaviyo.coresdk.model.ProfileKey.EXTERNAL_ID
+import com.klaviyo.coresdk.model.ProfileKey.PHONE_NUMBER
+import java.util.UUID
+
+/**
+ * Stores information on the currently active user
+ */
+internal object UserInfo {
+    var externalId: String = ""
+        set(value) {
+            field = value
+            Registry.dataStore.store(EXTERNAL_ID.name, value)
+        }
+        get() = field.ifEmpty {
+            field = Registry.dataStore.fetch(EXTERNAL_ID.name) ?: ""
+            field
+        }
+
+    var email: String = ""
+        set(value) {
+            field = value
+            Registry.dataStore.store(EMAIL.name, value)
+        }
+        get() = field.ifEmpty {
+            field = Registry.dataStore.fetch(EMAIL.name) ?: ""
+            field
+        }
+
+    var phoneNumber: String = ""
+        set(value) {
+            field = value
+            Registry.dataStore.store(PHONE_NUMBER.name, value)
+        }
+        get() = field.ifEmpty {
+            field = Registry.dataStore.fetch(PHONE_NUMBER.name) ?: ""
```

Comment:
Feels like a bit of duplicated code is there a way to maybe update this in one spot and then persist the whole thing or at least whatever changed since last time. Maybe though you are trying to optimize how often things are written to disk?

Reply:
Could you elaborate? Do you mean how each property is written to disk individually?

Reply:
@ndurell I made a change here to pull out some boilerplate and fix the fact that I'm storing an empty string instead of _clearing_ the key in the data store... did this address your comment at all too?

Reply:
I don't think so the main concern for me is it mixes couple the concerns of updating and getting state with persistence. I had thought in a previous chat you had mentioned something would observe changes to state and persist them. If we add a new field you will have to repeat this pattern again. I think though this class will probably not change very much over time so maybe it's not a big deal right now. Definitely not blocker!

Reply:
Ah I see what you're saying. Yeah when I realized there's only 4 things to persist about a profile, I didn't take on the added complication of a single state object that gets persisted altogether. I think if we later have more to persist in state, I'd consider changing this pattern.

Reply:
Resolving this per discussion on standup